server/api.go

```diff
@@ -136,13 +136,21 @@ func (h *Handler) Export(w http.ResponseWriter, r *http.Request) {
 	}
 
 	userID := r.Header.Get("Mattermost-User-ID")
-	channel, ok := h.hasPermissionToChannel(userID, channelID)
+	channel, ok := p.hasPermissionToChannel(userID, channelID)
 	if !ok {
 		handleError(w, http.StatusNotFound, "channel '%s' not found or user does not have permission", channelID)
 		return
 	}
 
-	postIterator := h.makePostsIterator(channel, showEmailAddress(h.client, userID))
+	conf := p.getConfiguration()
+	if conf.EnableAdminRestrictions {
+		if !(p.handler.client.User.HasPermissionTo(userID, model.PermissionManageChannelRoles) || p.handler.client.User.HasPermissionTo(userID, model.PermissionManageSystem)) {
+			handleError(w, http.StatusNotFound, "user does not have permission", channelID)
```

Comment:
nit: status code returned

```suggestion
			handleError(w, http.StatusForbidden, "user does not have permission", channelID)
```

Reply:
Good catch, should be fixed to `http.StatusForbidden`